### PR TITLE
dev/core#4567 Add mailing_id, is_test to civicrm_mailing_event_queue

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Add mailing_id, is_test to civicrm_mailing_event_queue

Before
----------------------------------------
No way to store link to mailing that does not rely on mailing_job records being retained

After
----------------------------------------
Field for mailing id created

Technical Details
----------------------------------------
Next step (once the upgrade script is added to this) is to start storing the mailing_id & is_test values ....

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by totten** [comment]
That's pretty clever how `updateMailingEventQueueTable()` optimizes the number `ALTER TABLE` operations.

I did some `r-run`, and it produced the new schema as expected.

The significant caveat (which Eileen alluded to on MM) is that... at some point, we'll start reading the new columns. So we will need to backfill these new columns. But it's OK to proceed with some other parts of the dev/core#4567 in parallel to that effort.

**Comment by eileenmcnaughton** [comment]
thanks @totten

File: CRM/Upgrade/Incremental/php/FiveSixtySeven.php [hunk 1]
```diff
@@ -36,6 +36,53 @@ public function upgrade_5_67_alpha1($rev): void {
     $this->addTask('Add cache_fill_took column to Group table', 'addColumn', 'civicrm_group', 'cache_fill_took',
       'DOUBLE DEFAULT NULL COMMENT "Seconds taken to fill smart group cache, not always related to cache_date"',
       FALSE);
+    $this->addTask('Update civicrm_mailing_event_queue to permit deleting records from civicrm_mailing_job', 'updateMailingEventQueueTable', 'updateMailingEventQueueTable');
```
Comment (colemanw) [inline CRM/Upgrade/Incremental/php/FiveSixtySeven.php:39]: @eileenmcnaughton this upgrade step has [reportedly caused someone's upgrader to crash](https://civicrm.stackexchange.com/questions/46045/error-executing-task-update-civicrm-mailing-event-queue-to-permit-deleting-reco). Any idea why?

Comment (eileenmcnaughton) [reply CRM/Upgrade/Incremental/php/FiveSixtySeven.php:39]: @colemanw table size I expect

**Pull request merged**